lib/Target/AMDGPU/SIFrameLowering.cpp

```diff
@@ -683,6 +683,12 @@ void SIFrameLowering::emitEpilogue(MachineFunction &MF,
       LivePhysRegs LiveRegs(*ST.getRegisterInfo());
       LiveRegs.addLiveIns(MBB);
 
+      // to avoid clobbering the registers used in the return instruction.
+      if (MBBI->getOpcode() == AMDGPU::S_SETPC_B64_return &&
```

Comment:
I assume it is safe to check for the opcode first, and then get the registers.
Not sure whether I should check the operand is register.